src/main/java/com/example/bookshop/service/BookServiceImpl.java

```diff
@@ -34,6 +34,29 @@ public BookDto getBookById(Long id) {
     @Override
     public BookDto createBook(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
-        return bookMapper.toDto(bookRepository.createBook(book));
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public BookDto updateBook(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
+
+        book.setTitle(requestDto.getTitle());
+        book.setAuthor(requestDto.getAuthor());
+        book.setIsbn(requestDto.getIsbn());
+        book.setPrice(requestDto.getPrice());
+        book.setDescription(requestDto.getDescription());
+        book.setCoverImage(requestDto.getCoverImage());
+
+        return bookMapper.toDto(bookRepository.save(book));
+    }
```

Comment:
I think you can change `id` in `book` to `id` from `requestDto`

Reply:
@HovorukhaBohdan Please, pay attention that CreateBookRequestDto doesn't have field `id`.